### PR TITLE
[9.0] [ADD] partner_sale_risk

### PR DESCRIPTION
**Pull request by carlosdauden**
This module extends #292

This PR will be rebased before merge

@Tecnativa

### CODE REVIEW

**Comment by yajo** [comment]
I'm afraid [you will have to make a quick fix to `base_partner_merge`](https://travis-ci.org/OCA/partner-contact/jobs/149926781#L447-L449) to get linters :green_heart:

**Comment by pedrobaeza** [comment]
Please rebase this one

**Comment by rafaelbn** [comment]
Rebuilding for testing

**Comment by andhit-r** [comment]
@carlosdauden i also encounter this PR unittest problem when i tried to backport it to 8.0. The problem lies on computation of **invoice_amount** and **invoice_pending** amount. The computation does not triggered when an invoice created. I tried to put depend on:
1. state
2. invoice_exists
3. invoice_ids
4. invoice_ids.amount_total
5. invoice_ids.invoice_line
6. order_line.invoice_lines.invoice_id.amount_total

Still no luck

File: partner_sale_risk/models/sale.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    def _compute_invoice_amount(self):
+        for order in self.filtered(lambda x: x.state == 'sale'):
+            order.invoice_amount = sum(
+                order.invoice_ids.mapped('amount_total'))
+            order.invoice_pending_amount = (
+                order.amount_total - order.invoice_amount)
+
+    @api.multi
+    def action_confirm(self):
+        if not self.env.context.get('bypass_risk', False):
```
Comment (pedrobaeza) [inline partner_sale_risk/models/sale.py]: Do it the contrary for avoiding one indentation level:
```
if self.env.context.get('bypass_risk'):
    return super(...)
partner = ...
```

File: partner_sale_risk/models/res_partner.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    risk_sale_order_include = fields.Boolean(
+        string='Include Sales Orders', help='Full risk computation')
+    risk_sale_order_limit = fields.Monetary(
+        string='Limit Sales Orders', help='Set 0 if it is not locked')
+    risk_sale_order = fields.Monetary(
```
Comment (pedrobaeza) [inline partner_sale_risk/models/res_partner.py:15]: Where is the currency_id field?

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
I miss here a risk check when validating an invoice

**Review by max3903 — APPROVED** [review]

File: partner_sale_risk/models/res_partner.py (hunk 2 above)
Comment (pedrobaeza) [reply partner_sale_risk/models/res_partner.py:15]: @carlosdauden please answer this

Comment (carlosdauden) [reply partner_sale_risk/models/res_partner.py:15]: Default currency_id field name is 'currency_id'

https://github.com/odoo/odoo/blob/aa09c522053b8c91dea557f9e9e71be2f4e965be/addons/account/models/partner.py#L414

https://github.com/odoo/odoo/blob/bbdf9605e4c480e1fc19b1faac9cfc4321bcaea2/openerp/fields.py#L1162-L1163

**Comment by carlosdauden** [comment]
@andhit-r can you test now?

**Comment by rafaelbn** [comment]
@pedrobaeza please review this PR you comments have been attended thanks

**Comment by rafaelbn** [comment]
@pedrobaeza you comment "I miss here a risk check when validating an invoice" is attended

**Review by rafaelbn — APPROVED** [review]
Tested in runbot functionally 👍

**Review by luismontalba — APPROVED** [review]
Function tested on runbot

**Pull request merged**